registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/SyncRegistrationServiceImpl.java

```diff
@@ -608,8 +688,64 @@ private RegistrationStatusDto convertEntityToDtoAndGetExternalStatus(SyncRegistr
 		return registrationStatusDto;
 	}
 
+	private String getPostalCode(String centerId, String langCode) {
+		String requestUrl = postalCodeReqUrl + "/" + centerId + "/" + langCode;
+		URI requestUri = URI.create(requestUrl);
+		String postalCode = null;
+		try {
+			String response = restApiClient.getApi(requestUri, String.class);
+			JSONObject jsonObjects = new JSONObject(response);
+			postalCode = ((JSONObject) jsonObjects.getJSONObject("response").getJSONArray("registrationCenters").get(0))
+					.getString("locationCode");
+		} catch (Exception e) {
+			regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
+					"", e.getMessage() + ExceptionUtils.getStackTrace(e));
+		}
+		return postalCode;
+	}
+
 	@Override
 	public boolean deleteAdditionalInfo(SyncRegistrationEntity syncEntity) {
 		return syncRegistrationDao.deleteAdditionalInfo(syncEntity);
 	}
+
+	private static byte[] getHMACHash(String value) throws java.security.NoSuchAlgorithmException {
+		if (value == null)
+			return null;
+		return HMACUtils2.generateHash(value.getBytes());
+	}
+
+	private static String getHMACHashCode(String value) throws java.security.NoSuchAlgorithmException {
+		if (value == null)
+			return null;
+		return CryptoUtil.encodeBase64(HMACUtils2.generateHash(value.getBytes()));
+	}
+
+	private String getHashCode(String value) {
+		StringBuilder hashedSaltValue = null;
+		try {
+			hashedSaltValue = new StringBuilder();
+			byte[] hashCode = getHMACHash(value);
+			byte[] nonce = Arrays.copyOfRange(hashCode, hashCode.length - 2, hashCode.length);
+			String result = convertBytesToHex(nonce);
+			Long hashValue = Long.parseLong(result, 16);
+			String salt=syncRegistrationDao.getSaltValue(hashValue);
+			for (int i = 0; i <= iteration; i++) {
+				hashedSaltValue = hashedSaltValue.append(getHMACHashCode(value + salt));

```

Comment:
This is not correct, you have to hash the result of hash for number of iterations

Reply:
Any update on this ? I still see the result of the hash is not hashed for iterations

Reply:
after discussion with taheer taken care

Reply:
I still think this is not taken care, 

hashedSaltValue += getHMACHash(saltHashValue); to be replaced with saltHashValue += getHMACHash(saltHashValue);

The result of the hash has to be hash again

Reply:
taken care